### PR TITLE
add weekly overview and option to save daily scores

### LINKED ISSUE #15: Add 7 daily quiz option to start page
## Value Proposition

- As a user,
- I want to be able to do quizzes on 7 days of a week and save my scores
- in order to have an overview over my scores and better monitor my eating habits

## Description

Image

## Acceptance criteria

- [x] On the startpage I have the option for monday till sunday to start a daily quiz
- [x] When the quiz is finished, I can restart the quiz and the current score is displayed below this button

## Tasks

- [x] Create branch
&nbsp;
- [x] Add headings with monday-sunday to startpage and a restart-quiz-button each, structure and style
- [x] Implement logic to save all scores to the local storage (probably save as an array)
- [x] Optionally create a reset button, to reset all quizzes at once

### PR DESCRIPTION
**Pull request by alinawinter**
fix #15

### CODE REVIEW

**Review by PReichetanz — CHANGES_REQUESTED** [review]
Really nice work! Don't worry about the number of comments, you've done great and my issues are only improvement ideas, keep going! 🚀

File: components/Form/FormPreselectionFood/index.js [hunk 1]
```diff
@@ -85,7 +85,7 @@ export default function FormPreselection({
       </CheckBoxList>
       <ButtonBox>
         <BasicButton
-          onClick={handleMarkAll}
+          onClick={handleSelectAll}
           aria-label="Auswahl markieren togglen"
```
[the remark this answers is not in the document]
Comment (alinawinter) [reply components/Form/FormPreselectionFood/index.js:89]: You're right, this is confusing... I changed it now to: `aria-label={
            isEverythingSelected
              ? "Kategorienauswahl zurücksetzen"
              : "Alle Kategorien auswählen"
          }`. I hope I got the idea right, how aria-labels are to be written for buttons

File: components/Form/FormPreselectionFood/index.js [hunk 2]
```diff
@@ -85,7 +85,7 @@ export default function FormPreselection({
       </CheckBoxList>
       <ButtonBox>
         <BasicButton
-          onClick={handleMarkAll}
+          onClick={handleSelectAll}
```
[the remark this answers is not in the document]
Comment (alinawinter) [reply components/Form/FormPreselectionFood/index.js:88]: nope, you're right, I added that

File: pages/scorepage/[weekday].js [hunk 3]
```diff
@@ -22,9 +32,9 @@ export default function ScorePage({
         <Score selectedFoodCategories={selectedFoodCategories} />
         <BasicButton
           onClick={handleSaveAndStartPage}
-          aria-label="Ergebnis speichern und zurück zur Startseite"
+          aria-label="Ergebnis speichern und zur Wochenübersicht"
```
[the remark this answers is not in the document]
Comment (alinawinter) [reply pages/scorepage/[weekday].js:35]: I removed the aria-label and added "button" as type (the data that is saved is not directly from a form), hope that's okay

File: components/Score/MiniScore.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+import { calculateAverageScoreAccordance } from "../../utils/averageScoreUtils";
+import { mergeArrayAllFoodAndSelectedFood } from "../../utils/mergeSelectedAndOtherFoodUtils";
+import {
+  calculateDetailsScoreDeviation,
+  calculateDetailsScoreAccordance,
+} from "../../utils/detailsScoreUtils";
+
+export default function MiniScore({ dailyQuizzesResultCollection, weekDay }) {
+  console.log("dailyQuizzesResultCollection", dailyQuizzesResultCollection);
+  const objectForWeekDay = dailyQuizzesResultCollection.find(
```
[the remark this answers is not in the document]
Comment (alinawinter) [reply components/Score/MiniScore.js:12]: you're right, I changed it to "dailyQuizForWeekDay"

File: components/Score/MiniScore.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
+import styled from "styled-components";
+import { foodCategories } from "../../lib/db";
```
[the remark this answers is not in the document]
Comment (alinawinter) [reply components/Score/MiniScore.js:2]: Okay, that's an interesting point! I implemented that here and in the other components that use it

File: components/WeekdayOptions/WeekdayOptions.js [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+  }
+
+  return (
+    <WeekdayOptionsCard
+      dailyQuizzesResultCollection={dailyQuizzesResultCollection}
+      weekDay={weekDay}
+    >
+      <DayAndButtonWrapper>
+        <h3>{weekDay}</h3>
+        <RestartQuizButton
```
[the remark this answers is not in the document]
Comment (alinawinter) [reply components/WeekdayOptions/WeekdayOptions.js:28]: Now it is :-D

File: components/WeekdayOptions/WeekdayOptions.js [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+          setSelectedFoodCategories={setSelectedFoodCategories}
+          weekDay={weekDay}
+          setCurrentWeekDay={setCurrentWeekDay}
+          setDailyQuizzesResultCollection={setDailyQuizzesResultCollection}
+          dailyQuizzesResultCollection={dailyQuizzesResultCollection}
+        />
+      </DayAndButtonWrapper>
+      {dailyQuizzesResultCollection.some(
+        (object) => object.weekday === weekDay
```
[the remark this answers is not in the document]
Comment (alinawinter) [reply components/WeekdayOptions/WeekdayOptions.js:37]: Changed it to "dailyQuiz"

File: components/Score/MiniScore.js [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+        {Math.floor(consumedQuantityAccordanceAverage)} %
+      </StyledScore>
+    </>
+  );
+}
+
+const StyledScore = styled.h1`
+  display: flex;
+  justify-content: center;
+  align-items: center;
```
[the remark this answers is not in the document]
Comment (alinawinter) [reply components/Score/MiniScore.js:40]: Yes, text-align doesn't completely center the h1 into the circle. If I still have time later to go through everything, I'm thinking about making a div that I place the h1 into instead... and maybe refactoring it with the score.js

File: components/Score/MiniScore.js [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+  color: var(--color-blue);
+  font-family: var(--font-family-text);
+  font-size: 1em;
+  background-color: var(--color-yellow);
+  border: none;
+  border-radius: 50%;
+  width: 3.5em;
+  height: 3.5em;
+  padding: 0.1em 0.1em;
+  cursor: pointer;
```
[the remark this answers is not in the document]
Comment (alinawinter) [reply components/Score/MiniScore.js:50]: Yes, it's a mistake, I copy pasted my button styles and just changed some things, forgot to delete that as well

File: pages/index.js [hunk 10]
```diff
@@ -1,65 +1,40 @@
...
+  ];
+
+  console.log(dailyQuizzesResultCollection);
+
   return (
     <Layout>
-      <StartContentWrapper>
-        <RestartQuizButton
+      <h2>Deine Woche</h2>
+      {weekDays.map((weekDay) => (
```
[the remark this answers is not in the document]
Comment (alinawinter) [reply pages/index.js:27]: Oh, thats a really good hint, I implemented that!

File: components/WeekdayOptions/WeekdayOptionsCard.js [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+      dailyQuizzesResultCollection={dailyQuizzesResultCollection}
+      weekDay={weekDay}
+    >
+      {children}
+    </StyledContainer>
+  );
+}
+
+const StyledContainer = styled.div`
+  background-color: ${({ dailyQuizzesResultCollection, weekDay }) =>
```
[the remark this answers is not in the document]
Comment (alinawinter) [reply components/WeekdayOptions/WeekdayOptionsCard.js:19]: I actually had such a quizTaken-variable in another component, which I now uplifted to the parent component and handed down! Thanks alot!

File: pages/_app.js [hunk 12]
```diff
@@ -65,6 +66,8 @@ export default function App({ Component, pageProps }) {
         handleSetQuantityPerCategory={handleSetQuantityPerCategory}
         dailyQuizzesResultCollection={dailyQuizzesResultCollection}
         setDailyQuizzesResultCollection={setDailyQuizzesResultCollection}
+        currentWeekDay={currentWeekDay}
+        setCurrentWeekDay={setCurrentWeekDay}
```
[the remark this answers is not in the document]
Comment (alinawinter) [reply pages/_app.js:70]: I had completely forgot about this topic... I implemented that now for setCurrentWeekDay and setDailyQuizzesResultCollection. I have it in another parts of my code like this as well I think, which I will take care of in another PR.

**Comment by alinawinter** [comment]
Thanks alot @PReichetanz ! It is very much appreciated! :-) I integrated all your feedback

File: pages/index.js [hunk 13]
```diff
@@ -19,22 +20,33 @@ export default function Home({
...
-          key={weekDay}
-          setSelectedFoodCategories={setSelectedFoodCategories}
-          dailyQuizzesResultCollection={dailyQuizzesResultCollection}
-          setDailyQuizzesResultCollection={setDailyQuizzesResultCollection}
-          weekDay={weekDay}
-          setCurrentWeekDay={setCurrentWeekDay}
-          currentWeekDay={currentWeekDay}
-        />
-      ))}
+      <List>
```
Comment (PReichetanz) [inline pages/index.js:26]: To ensure a11y in Safari as well because you've set `list-style: none`;
```suggestion
      <List role="list">
```

**Review by PReichetanz — APPROVED** [review]

File: pages/index.js (hunk 13 above)
Comment (alinawinter) [reply pages/index.js]: integrated that now, thank you!!

**Pull request merged**